src/Utilities.ts

```diff
@@ -327,32 +308,66 @@ export class Utilities {
      * Handles cleaning up the Sarif Viewer temp directory used for temp files (embedded code, converted files, etc.)
      */
     public static removeSarifViewerTempDirectory(): void {
-        const path = Utilities.Path.join(Utilities.Os.tmpdir(), Utilities.SarifViewerTempDir);
-        Utilities.removeDirectoryContents(path);
-        Utilities.Fs.rmdirSync(path);
+        const tempPath = path.join(Utilities.Os.tmpdir(), Utilities.SarifViewerTempDir);
+        Utilities.removeDirectoryContents(tempPath);
+        fs.rmdirSync(tempPath);
+    }
+
+    /**
+     * Fixes up file schemed based paths to contain the proper casing for VSCode's URIs which are case sensitive.
+     * @param uri The URI for which to fix the casing.
+     */
+    public static fixUriCasing(uri: Uri): Uri {
+        // We can only support file-system scheme files.
+        if (uri.scheme.toUpperCase() !== "FILE") {
+            return uri;
+        }
+
+        let pathPartIndex: number = 0;
+        const pathParts: string[] = uri.fsPath.split(path.sep);
+        if (pathParts.length < 1) {
+            return uri;
+        }
+
+        // We assume that the "root" drive is lower-cased which is "usually" the case.
```

Comment:
>. [](start = 85, length = 1)

I understand that by lower-casing the drive, you avoid duplicates ("[file:///C:/file.c](file:///C:/file.c)" vs. "[file:///c:/file.c](file:///c:/file.c)"). So I agree with the line of code. But is the comment right? In what was does forcing the drive letter to lower case embody an assumption?